Build failure: tic-80 (CMake 4 error)

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

```sh
nix build 'nixpkgs#tic-80'
```
_or_
```sh
nix-build '<nixpkgs>' -A tic-80
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/309187927

### Relevant log output

```console
[…]
-- Checking for module 'zlib'
--   Found zlib, version 1.3.1
-- Use system library: zlib
CMake Warning at cmake/zip.cmake:17 (message):
  System library kubazip not found
Call Stack (most recent call first):
  CMakeLists.txt:154 (include)


CMake Error at vendor/zip/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!
```

### Additional context

See #445447, as well as the upstream issue <https://github.com/nesbox/TIC-80/issues/2823>.

TIC-80's own `CMakeLists.txt` is fine. The problem is from the [`zip` library](https://github.com/kuba--/zip) it uses (submoduled as `vendor/zip`). That library's upstream actually bumped its CMake version [several years ago](https://github.com/kuba--/zip/commit/a597bdf814684a5a8b4adbae8597f03c865601fd), but the submodule is still on [an even earlier commit](https://github.com/kuba--/zip/commit/d7df626f3aa457e01669f65e61bf8f484e352941).

### System metadata

 - system: `"x86_64-darwin"`
 - host os: `Darwin 24.5.0, macOS 15.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/n4vf8ia28wa77g29cjmg5jn3516qp5qx-nixpkgs/nixpkgs`

### Notify maintainers

`tic-80`: @blinry

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ku/kubazip/package.nix
+++ b/pkgs/by-name/ku/kubazip/package.nix
@@ -1,0 +1,44 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  cmake,
+  ninja,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "kubazip";
+  version = "0.3.5";
+
+  src = fetchFromGitHub {
+    owner = "kuba--";
+    repo = "zip";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-wNkIYuwwXo7v3vNaOnRZt1tcd0RGjDvCUqDGdvJzVdo=";
+  };
+
+  outputs = [
+    "out"
+    "dev"
+  ];
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeBool "BUILD_SHARED_LIBS" (!stdenv.hostPlatform.isStatic))
+  ];
+
+  doCheck = true;
+
+  meta = {
+    description = "Portable, simple zip library written in C";
+    homepage = "https://github.com/kuba--/zip";
+    changelog = "https://github.com/kuba--/zip/releases/tag/v${finalAttrs.version}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ marcin-serwin ];
+  };
+})

--- a/pkgs/by-name/ti/tic-80/package.nix
+++ b/pkgs/by-name/ti/tic-80/package.nix
@@ -7,6 +7,7 @@
   rake,
   curl,
   fetchFromGitHub,
+  kubazip,
   libGL,
   libGLU,
   libX11,
@@ -37,7 +38,7 @@ stdenv.mkDerivation {
     hash = "sha256-UjBnXxYZ5gfk58sI1qek5fkKpJ7LzOVmrxdjVgONcXc=";
     # TIC-80 vendors its dependencies as submodules. For the following dependencies,
     # there are no (or no compatible) packages in nixpkgs yet, so we use the vendored
-    # ones as a fill-in: kubazip, wasm, squirrel, pocketpy, argparse, naett,
+    # ones as a fill-in: wasm, squirrel, pocketpy, argparse, naett,
     # sdlgpu, mruby.
     fetchSubmodules = true;
   };
@@ -90,6 +91,7 @@ stdenv.mkDerivation {
     rake
   ];
   buildInputs = [
+    kubazip
     libGL
     libGLU
     libX11

